frontend/packages/console-shared/src/components/formik-fields/InputGroupField.tsx

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable no-unused-vars, no-undef */
+import * as React from 'react';
+import { InputGroup, TextInput, TextArea, TextInputTypes } from '@patternfly/react-core';
+import { GroupInputProps, BaseInputFieldProps, TextType } from './field-types';
+import BaseInputField from './BaseInputField';
+
+export type InputGroupFieldProps = GroupInputProps &
+  BaseInputFieldProps & {
+    type?: TextInputTypes;
```

Comment:
When type is needed in both `InputField` and `InputGroupField` then why did you remove it from `BaseInputField` and adding it here explicitly?

Reply:
@rohitkrai03  The reason behind doing this is, for InputField I want to keep the `type` mandatory & for `InputGroupField` want it optional because users can have both TextArea and Text and as we know for TextArea user don't need type. Let me know if there's a better way to do this

Reply:
You can make type optional with having text as a default for InputField. Else just make use of the extend mechanism in the field-types file like all the other props defined there.

Reply:
made the `type` optional with text as a default for InputField